arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -87,36 +86,40 @@ class CreateSymbolStylesFromWebStylesViewController: UIViewController, UIAdaptiv
     /// - Parameters:
     ///   - symbolStyle: An `AGSSymbolStyle` object.
     ///   - types: The types of symbols to search in the symbol style.
-    ///   - completion: A closure executed upon success.
-    private func getSymbols(symbolStyle: AGSSymbolStyle, types: [SymbolType], completion: @escaping () -> Void) {
+    private func getSymbols(symbolStyle: AGSSymbolStyle, types: [SymbolType]) {
         let getSymbolsGroup = DispatchGroup()
         var legendItems = [LegendItem]()
-        var symbols = [(String, AGSSymbol, [String])]()
+        var symbolInfos = [AGSSymbol: (String, [String])]()
         
+        // Get the `AGSSymbol` for each symbol type.
         types.forEach { category in
             getSymbolsGroup.enter()
             let symbolName = category.symbolName
             let symbolCategoryValues = category.symbolCategoryValues
             symbolStyle.symbol(forKeys: [symbolName]) { symbol, _ in
-                // Add the symbol to result collections and ignore any error.
-                if let symbol = symbol {
-                    symbols.append((symbolName, symbol, symbolCategoryValues))
-                    symbol.createSwatch { image, _ in
-                        defer {
-                            getSymbolsGroup.leave()
-                        }
-                        if let image = image {
-                            legendItems.append(LegendItem(name: symbolName, image: image))
-                        }
-                    }
+                defer { getSymbolsGroup.leave() }
+                // Add the symbol to result dictionary and ignore any error.
+                guard let symbol = symbol else { return }
+                symbolInfos[symbol] = (symbolName, symbolCategoryValues)
+                
+                // Get the image swatch for the symbol.
+                getSymbolsGroup.enter()
+                symbol.createSwatch { image, _ in
+                    defer { getSymbolsGroup.leave() }
```

Comment:
Enter the dispatch group twice for each symbol, to ensure we don't get stuck if any error occurs.